src/Infrastructure/Cli/Ssh.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Akeneo\PimMigration\Infrastructure\Cli;
+
+use Akeneo\PimMigration\Infrastructure\ImpossibleConnectionException;
+
+/**
+ * @author    Philippe Mossière <philippe.mossiere@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ */
+class Ssh
+{
+    /** @var string */
+    private $host;
+
+    /** @var int */
+    private $port;
+
+    /** @var resource */
+    private $connection = false;
+
+    public function __construct(string $host, int $port = 22)
+    {
+        $this->host = $host;
+        $this->port = $port;
+    }
+
+    public function connect()
+    {
+        $this->connection = ssh2_connect($this->host, $this->port);
+        $this->checkConnection();
+
+        return $this->connection;
+    }
+
+    public function login(string $username): void
+    {
+        $this->checkConnection();
+
+        if (false === ssh2_auth_agent($this->connection, $username)) {
+            throw new ImpossibleConnectionException(
+                sprintf(
+                    'Impossible to login to %s@%s:%d using ssh local agent, try to run ssh-add before retry',
+                    $username,
+                    $this->host,
+                    $this->port
+                )
+            );
+        }
+    }
+
+    public function exec(string $command): ?string
```

Comment:
Why not return a SshResult object?

Reply:
Ideally, it would have the content and the return code in this object. But pretty hard/dirty: https://stackoverflow.com/questions/10478491/php-ssh2-exec-channel-exit-status